clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -46,13 +48,13 @@
 public class SubscriptionState {
 
     /* the list of topics the user has requested */
-    private final Set<String> subscribedTopics;
+    private final Set<String> subscription;
 
     /* the list of partitions the user has requested */
-    private final Set<TopicPartition> subscribedPartitions;
+    private final Set<TopicPartition> userAssignment;
```

Comment:
Not sure I understand the question. Are you suggesting reuse of the assignment collection below? I think the only problem with that is that we use this collection to detect when a user is trying to mix topic subscription and manual assignment. We could probably also do this just with a flag though.